style_guide.rst

```diff
@@ -0,0 +1,393 @@
+and always by a small number of additional characters. Also, remember that
+most syntactical elements in C are delimited by whitespace, and that
+newlines are considered whitespace. So, the following is correct C::
+
+    if (VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_LONG_CONDITION ||
+        ANOTHER_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_VERY_LONG_CONDITION ||
+        strlen(s) < 100)
+    {
+        /* ... */
+    }
+
+Blank Lines
+^^^^^^^^^^^
+
+Separate function definitions with two blank lines.
+
+Extra blank lines may be used (sparingly) to separate groups of
+related functions.
+
+Use blank lines in functions, sparingly, to indicate logical sections.
+
+
+Whitespace in Expressions and Statements
+----------------------------------------
+
+Avoid extraneous whitespace in the following situations:
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Immediately inside parentheses:
+
+  ::
+
+    Yes: chilog(INFO, "User %s has connected", user);
+    No:  chilog( INFO, "User %s has connected", user );
+
+- Immediately before a comma or a semicolon:
+
+  ::
+
+    Yes: for (int i = 0; i < N; i++)
+    No:  for (int i = 0 ; i < N ; i++)
+
+  ::
+
+    Yes: chilog(INFO, "User %s has connected", user);
+    No:  chilog(INFO , "User %s has connected" , user);
+
+- Immediately before the open parenthesis that starts the argument list of a function call:
+
+  ::
+
+    Yes: chilog(INFO, "User %s has connected", user);
+    No:  chilog (INFO, "User %s has connected", user);
+
+- Immediately before brackets indicating an array index:
+
+  ::
+
+    Yes: a[N] = b[i]
+    No:  a [N] = b [i]
+
+- More than one space around an assignment (or other) operator to align it with another.
+
+  Yes:
+
+  ::
+
+       x = 1;
+       y = 2;
+       long_variable = 3;
+
+  No:
+
+  ::
+
+       x             = 1
+       y             = 2
+       long_variable = 3
+
+
+Places to use spaces
+^^^^^^^^^^^^^^^^^^^^
+
+- Always surround these binary operators with a single space on either side: assignment (=), augmented assignment (+=, -=, etc.), comparisons (==, <, >, !=, <>, <=, >=), Booleans (&&, ||).
+
+- Use a space after these keywords::
+
+	if, switch, case, for, do, while
+
+  Yes:
+
+  ::
+
+       if (x == 5)
+
+  No:
+
+  ::
+
+       if(x == 5)
+
+- Use spaces around arithmetic operators:
+
+  Yes:
+
+  ::
+
+        i = i + 1
+        submitted += 1
+        x = x * 2 - 1
+        hypot2 = x * x + y * y
+        c = (a + b) * (a - b)
+
+  No:
+
+  ::
+
+        i=i+1
+        submitted +=1
+        x = x*2 - 1
+        hypot2 = x*x + y*y
+        c = (a+b) * (a-b)
+
+- Compound statements (multiple statements on the same line) are generally discouraged.
+
+
+  Yes:
+
+  ::
+
+        if (x == 0)
+            do_blah_thing();
+        do_one();
+        do_two();
+        do_three();
+
+  Rather not:
+
+  ::
+
+        if (x == 0) do_blah_thing();
+        do_one(); do_two(); do_three();
+
```

Comment:
Might be worth explicitly calling out spacing around "->"

Reply:
Added some text on the dereference, dot, and arrow operators.